addons/resource/models/resource_resource.py

```diff
@@ -144,6 +144,8 @@ def _get_unavailable_intervals(self, start, end):
             calendar_mapping[resource.calendar_id or resource.company_id.resource_calendar_id] |= resource
```

Comment:
Let's see what @tivisse thinks or someone in HR Team. 🙂 

I think we could avoid filling that dict if we know the key will be `False` since we will do nothing with the value for that key. 🤷🏼‍♂️

Reply:
@xavierbol You mean by doing something like this?
```py
if not resource.calendar_id and not resource.company_id.calendar_id:
    continue
```
If yes, it can indeed make sense, making the condition below obsolete. I'm not sure if there is a lot of difference in performance though :thinking: